infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/SessionManager.java

```diff
@@ -182,8 +198,8 @@ public synchronized void refresh(@Nonnull TopologyDescriptor newTopology) {
                 metadataManager.setTopologyConfigId(topology.getTopologyConfigId());
 
                 stopReplication(sessionsToRemove);
-
-                updateReplicationParameters(sessionsUnchanged);
+                updateReplicationParameters(Sets.intersection(sessionsUnchanged, outgoingSessions));
+                router.updateTopologyConfigId(topology.getTopologyConfigId());
```

Comment:
I thought we need to find a way to kind of centralize the in-memory topology, since it seems like we are spreading it in multiple places (SessionManager, LogReplicationContext, router, SinkManager, etc) some of them just need the topology config id but still it could get from topology

maybe `LogReplicationContext` or `SessionManager` is a good place?

Reply:
Thanks! I'll reevaluate this..